virtool/groups/data.py

```diff
@@ -47,17 +52,32 @@ async def find(self) -> List[GroupMinimal]:
             async for document in self._mongo.groups.find()
         ]
 
-    async def get(self, group_id: str) -> Group:
+    async def get(self, group_id: str | int) -> Group:
```

Comment:
I think this function can be simplified. I would divide it into two main cases:

1. the id is an int -> Then you know the group must be on postgres as int ids mean nothing to mongo. Raise an error if no group is found
2. The id is a string -> then you need to check both DBs. I would rewrite so it fetches the group from both DBs at the same time along with the users list